contracts/Removal.sol

```diff
@@ -383,42 +380,67 @@ contract Removal is ERC1155SupplyUpgradeable, PausableAccessPreset {
     return super.supportsInterface(interfaceId);
   }
 
+  // todo improve docs to describe the order of which things are released from (unlisted -> market -> certificate)
   /**
    * @notice Burns an amount of tokens from an account.
-   * @dev Destroys `amount` tokens of token type `id` from `from`. If the tokens are included as part of any
+   * @dev Destroys `amount` tokens of token type `id`. If the tokens are included as part of any
    * certificates, the certificate balances are decremented by `amount` as well.
    *
    * ##### Requirements:
    *
    * - Enforces the rules of `ERC1155Upgradeable._burn`.
    */
-  function _burn(
-    address from,
-    uint256 id,
-    uint256 amount
-  ) internal override {
-    if (from == address(_market)) {
-      _market.release(id, amount);
-    } else if (from == address(_certificate)) {
-      Certificate.Balance[] memory certificatesOfRemoval = _certificate
-        .certificatesOfRemoval(id);
-      uint256 numberOfCertificatesForRemoval = certificatesOfRemoval.length;
-      bytes[] memory releaseCalls = new bytes[](numberOfCertificatesForRemoval);
-      for (uint256 i = 0; i < numberOfCertificatesForRemoval; ++i) {
-        // todo releaseFromCertificate vs releaseUnlisted
-        Certificate.Balance memory certificateBalance = certificatesOfRemoval[
-          i
-        ];
-        releaseCalls[i] = abi.encodeWithSelector(
-          _certificate.releaseRemoval.selector,
-          certificateBalance.id,
-          id,
-          certificateBalance.amount
+  function _burn(uint256 id, uint256 amount) internal {
+    // todo needs more tests
+    uint256 amountBurned = 0;
+    address market = this.marketAddress();
+    address certificate = this.certificateAddress();
+    uint256 unlistedBalance = balanceOf(RemovalIdLib.supplierAddress(id), id);
+    uint256 listedBalance = balanceOf(market, id);
+    uint256 soldBalance = balanceOf(certificate, id);
+    if (unlistedBalance > 0) {
+      amountBurned += MathUpgradeable.min(amount, unlistedBalance);
+      super._burn(RemovalIdLib.supplierAddress(id), id, amountBurned); // todo single call to _burnBatch or emit event
+    }
+    if (amount - amountBurned <= amount) {
+      if (listedBalance > 0) {
```

Comment:
```suggestion
      if (amountBurned < amount && listedBalance > 0) {
```

Reply:
this should be addressed [here](https://github.com/nori-dot-eco/contracts/pull/306/files#diff-1f1e10ef0e300f31ed72a37f4be0f5ec9ed7512c3ab3a09020b43f623913e6bfR402) now. Had wrong checks almost entirely in this PR